Tweak docker entrypoint to support regtest, testnet

joemphilips opened the pull request:
Before this, docker image will never detects that
`lightning-rpc` was created if it is running in regtest
or testnet, because the file will be created under
subfolder for each network name.
~By recursively checking the file creation, now docker
can be run in any network configuration.~
I had to deal the path for argument to the `socat` , so I added a new `ENV` entry in Dockerfile to detect the correct network.

ZmnSCPxj commented on tools/docker-entrypoint.sh:
```diff
@@ -2,18 +2,20 @@
 
 : "${EXPOSE_TCP:=false}"
 
+datadir="${LIGHTNINGD_DATA}/${LIGHTNINGD_NETWORK}"
```
Suggest renaming to `networkdatadir`.

ZmnSCPxj commented on tools/docker-entrypoint.sh:
```diff
@@ -2,18 +2,20 @@
     fg %-
 else
-    exec lightningd "$@"
+    exec lightningd --network=${LIGHTNINGD_NETWORK} "$@"
```
`shellcheck` wants quote marks around the variable.

ZmnSCPxj commented:
Also commit title should probably change as well.

joemphilips commented:
Updated according to the review. I also dropped `-r` option in`docker-entrypoint.sh` since it is unnecessary after I introduced `LIGHTNING_NETWORK` env variable.

joemphilips commented:
> Should not commit message say "take LIGHTNING_NETWORK env variable in Dockerfile"?

Right, I was careless. updated.

ZmnSCPxj commented:
Also probably deserves a Changelog entry in the commit message:

    Changelog-Added: Docker build now includes a `LIGHTNINGD_NETWORK` variable you can override to `regtest` or `testnet` or any valid argument to `--network`.

Though wording might be changed depending on how exactly Docker terminology is, I am not a Docker expert.

joemphilips commented:
Changelog added. I couldn't find any past changelog which uses newline. So I avoided using it.

ZmnSCPxj commented:
Rebased on master.  Reapplying ACK:

ACK 5092f334

cdecker commented:
ACK https://github.com/ElementsProject/lightning/pull/3813/commits/5092f3342cc8b8f54f4efc23f9e8e86787992ad1

ZmnSCPxj commented on tools/docker-entrypoint.sh:
```diff
@@ -2,18 +2,20 @@
     echo "C-Lightning starting"
-    while read -r i; do if [ "$i" = "lightning-rpc" ]; then break; fi; done \
-    < <(inotifywait  -e create,open --format '%f' --quiet "$LIGHTNINGD_DATA" --monitor)
+    while read i; do if [ "$i" = "lightning-rpc" ]; then break; fi; done \
```
Hmm, this seems to be different.  `-r` for `read` means "ignore backslashes on input", which is recommended.  `shellcheck` is also complaining about this as well.  Maybe should not be removed?

joemphilips replied:
Yes, it is my fault. fixed.
